samples/siwx91x_ota/src/main.c

```diff
@@ -0,0 +1,582 @@
+		.psk = (uint8_t *)ctx->wifi_psk,
+		.ssid_length = strlen(ctx->wifi_ssid),
+		.ssid = (uint8_t *)ctx->wifi_ssid
+	};
+	struct net_if *iface = net_if_get_first_wifi();
+	int ret;
+
+	ret = net_mgmt(NET_REQUEST_WIFI_CONNECT, iface, &cnx_params,
+		       sizeof(struct wifi_connect_req_params));
+	if (ret) {
+		printf("Connection request failed: %s\n", strerror(-ret));
+	}
+	return ret;
+}
+
+/*
+ * @brief Wait for IP configuration to complete
+ */
+static int ota_wait_for_ip_config(struct app_ctx *ctx)
+{
+	int ret;
+
+	if (IS_ENABLED(CONFIG_NET_IPV4)) {
+		ret = k_event_wait_safe(&ctx->events, NET_IPV4_EVENT_READY, false, K_SECONDS(30));
+	} else {
+		ret = k_event_wait_safe(&ctx->events, NET_IPV6_EVENT_READY, false, K_SECONDS(30));
+	}
+	if (!ret && IS_ENABLED(CONFIG_NET_IPV6)) {
+		ret = k_event_wait_safe(&ctx->events, NET_IPV6_EVENT_READY, false, K_NO_WAIT);
+	}
+	return (ret > 0) ? 0 : -ETIMEDOUT;
+}
+
+/*
+ * @brief Connect to OTA server
+ */
+static int ota_connect_to_server(struct app_ctx *ctx)
+{
+	int peer_verify = TLS_PEER_VERIFY_NONE;
+	struct zsock_addrinfo *res;
+	int ret;
+
+	ret = zsock_getaddrinfo(ctx->url.host, ctx->url.port, NULL, &res);
+	if (ret != 0) {
+		printf("Cannot resolve %s:%s: %s\n", ctx->url.host, ctx->url.port,
+		       zsock_gai_strerror(ret));
+		return -EINVAL;
+	}
+
+	if (ctx->url.use_tls) {
+		ctx->sock = zsock_socket(res->ai_family, SOCK_STREAM, IPPROTO_TLS_1_2);
+		if (ctx->sock < 0) {
+			goto error;
+		}
+		ret = zsock_setsockopt(ctx->sock, SOL_TLS, TLS_PEER_VERIFY, &peer_verify,
+				       sizeof(peer_verify));
+		if (ret < 0) {
+			goto error;
+		}
+	} else {
+		/* Use regular TCP for HTTP */
+		ctx->sock = zsock_socket(res->ai_family, SOCK_STREAM, IPPROTO_TCP);
+		if (ctx->sock < 0) {
+			goto error;
+		}
+	}
+
+	/* Connect to the server */
+	ret = zsock_connect(ctx->sock, res->ai_addr, res->ai_addrlen);
+	if (ret < 0) {
+		printf("Connection failed: %s\n", strerror(errno));
+		goto error;
+	}
+
+	printf("Connected to %s:%s\n", ctx->url.host, ctx->url.port);
+	zsock_freeaddrinfo(res);
+	return 0;
+
+error:
+	zsock_close(ctx->sock);
+	ctx->sock = -1;
+	zsock_freeaddrinfo(res);
+	return -errno;
+}
+
+/*
+ * @brief HTTP response callback - processes firmware chunks
+ */
+static int ota_http_response_cb(struct http_response *rsp, enum http_final_call final_data,
+				void *user_data)
+{
+	struct app_ctx *ctx = (struct app_ctx *)user_data;
+
+	ctx->http_status_code = rsp->http_status_code;
+	ctx->image_size = rsp->content_range.total;
+
+	if (ctx->flash_buffer_len + rsp->body_frag_len > sizeof(ctx->flash_buffer)) {
+		return -ENODATA;
+	}
+	memcpy(ctx->flash_buffer + ctx->flash_buffer_len, rsp->body_frag_start, rsp->body_frag_len);
+	ctx->flash_buffer_len += rsp->body_frag_len;
+
+	return 0;
+}
+
+/*
+ * @brief Download firmware chunk from server
+ */
+static int ota_download_firmware_chunk(struct app_ctx *ctx)
+{
+	char range_header[64];
+	const char *headers[] = { range_header, NULL };
```

Comment:
The compliance Ci want to see this array declared with `static` (or you can remove the `const` I assume)